components/timezone/src/ids.rs

```diff
@@ -66,6 +68,7 @@ impl TimeZoneIdMapper {
     /// ✨ *Enabled with the `compiled_data` Cargo feature.*
     ///
     /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[allow(clippy::new_without_default)] // feature-gated constructor
```

Comment:
also feature-gate `Default`? I think we do this in other places.

Reply:
Ok I found precedent for this in `GraphemeClusterSegmenter`.